alfresco-integration/shared/src/main/java/com/github/dynamicextensionsalfresco/osgi/BundleDependencies.java

```diff
@@ -0,0 +1,125 @@
+package com.github.dynamicextensionsalfresco.osgi;
+
+import com.springsource.util.osgi.manifest.BundleManifest;
+import com.springsource.util.osgi.manifest.BundleManifestFactory;
+import com.springsource.util.osgi.manifest.ExportedPackage;
+import com.springsource.util.osgi.manifest.ImportedPackage;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.List;
+import kotlin.jvm.internal.Intrinsics;
+import org.jetbrains.annotations.NotNull;
+import org.osgi.framework.Bundle;
+
+/**
+ * Sort a list of bundles by their dependency graph. Bundles without or satisfied dependencies come first.
+ *
+ * @author Laurent Van der Linden.
+ * @author Toon Geens
+ */
+public class BundleDependencies {
```

Comment:
@rschev Can you shine your light on the conversion of this "kotlin object" to java ?

Reply:
Looks fine to me. The BundleDependencies object was mostly just used as a stateless container for functions, so a class with a few static methods seems like a good translation to me.

Reply:
Given this class is a stateless container, I would propose to give it a private no-arg constructor and making the class final